test/hack/cleanup/main.go

```diff
@@ -93,3 +105,26 @@ func main() {
 		}
 	}
 }
+
+func cleanUpCluster(ctx context.Context, name string, client *eks.Client, log *zap.SugaredLogger, expirationTime time.Time) {
+	if name != "" {
+		if strings.HasPrefix(name, "soak-") {
+			cluster, err := client.DescribeCluster(ctx, &eks.DescribeClusterInput{Name: aws.String(name)})
+			if err != nil {
+				log.Errorf("%v", err)
+			}
+
+			if !lo.FromPtr(cluster.Cluster.CreatedAt).Before(expirationTime) {
+				log.Infof("soak testing cluster (%s) will not need to be cleanup", name)
+				os.Exit(0)
+			}
+		}
+
+		deleteCluster := exec.Command("eksctl", "delete", "cluster", "--name", name, "--timeout", "60m", "--wait")
+		fmt.Println(deleteCluster.String())
+		if out, err := deleteCluster.Output(); err != nil {
+			fmt.Println(string(out))
+			log.Errorf("%v", err)
+		}
```

Comment:
Will this always work? I feel like we probably need our cleanup logic to refer to the cloud formation stacks so that it cleans up all associate resources too.

Reply:
So EKSCTL does cleanup all the stacks. EKSCTL is aware of the dependency between stacks. For example, we will need to delete the stack that creates the NodeGroups first and wait for that to be deleted completely before we can delete the cluster. I think it would be cleaner to allow EKSCTL to handle that logic.

Reply:
Ah you're right, got my wires crossed. Still, this gets me thinking, what's the retry rate on this? Just the recurrence of the cron?